mlflow/tensorflow/__init__.py

```diff
@@ -805,7 +856,37 @@ def log_model_without_starting_new_run():
                         tf_meta_graph_tags=[tag_constants.SERVING],
                         tf_signature_def_key="predict",
                     )
-                    save_model(path=local_path, mlflow_model=mlflow_model, **save_model_kwargs)
+
+                    input_example = None
+                    signature = None
+                    if log_models:
+
+                        def predict_input_fn():
+                            input_slices = input_example_slice
+                            if isinstance(input_example_slice, dict):
+                                input_slices = {
+                                    k: tensorflow.convert_to_tensor(v)
+                                    for k, v in input_example_slice.items()
+                                }
+                            return tensorflow.data.Dataset.from_tensor_slices(input_slices).batch(1)
+
+                        predicted_values = [v for v in tf_model.predict(predict_input_fn)]
```

Comment:
What does this accomplish?

Reply:
done and added a docstring to the predict input function. As discussed will add examples to the methods to make it easier to understand, but there are unit tests for all 'x' types that document the behavior